src/petro/extractor/looper_layer.hpp

```diff
@@ -32,16 +32,22 @@ class looper_layer : public Super
         m_loop = false;
     }
 
-    /// Returns the samples timestamp, offset based on the looping
-    uint64_t timestamp() const
+    /// Returns the sample's decoding timestamp with the loop offset
+    uint64_t decoding_timestamp() const
     {
-        return m_loop_offset + Super::timestamp();
+        return m_loop_offset + Super::decoding_timestamp();
+    }
+
+    /// Returns the sample's presentation timestamp with the loop offset
+    uint64_t presentation_timestamp() const
+    {
+        return m_loop_offset + Super::presentation_timestamp();
     }
 
     /// Advances to the next sample
     void advance()
     {
-        auto timestamp = Super::timestamp();
+        auto timestamp = Super::decoding_timestamp();
```

Comment:
We cannot use the presentation timestamp for this, because it is not monotonic, alhtough it is possible that two timestamps are the same for the very last frame. Another solution would be using the actual length of the video/audio track as the loop offset. But I'm not certain that we get the same values for video and audio :(
I tested the current solution with a lot of different videos and the looping worked nicely.